File: src/main/java/com/timeback/ui/feature/goals/GoalsViewModel.java
package com.timeback.ui.feature.goals;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

import com.timeback.ui.domain.gateway.FeatureGateway;
import com.timeback.ui.domain.model.ActionResult;
import com.timeback.ui.domain.model.Duration;
import com.timeback.ui.domain.model.ScreenState;

import javax.inject.Inject;
import dagger.hilt.android.lifecycle.HiltViewModel;

/**
 * UI-06 목표 ViewModel.
 * 목표 목록 조회 및 새 목표 생성을 처리한다.
 */
@HiltViewModel
public class GoalsViewModel extends ViewModel {

    private final FeatureGateway gateway;
    private final MutableLiveData<ScreenState> _state = new MutableLiveData<>(new ScreenState.Initial());
    public final LiveData<ScreenState> state = _state;

    private final MutableLiveData<ActionResult> _actionResult = new MutableLiveData<>();
    public final LiveData<ActionResult> actionResult = _actionResult;

    @Inject
    public GoalsViewModel(FeatureGateway gateway) {
        this.gateway = gateway;
        loadGoals();
    }

    public void loadGoals() {
        _state.setValue(new ScreenState.Loading());
        gateway.readGoals().thenAccept(_state::postValue);
    }

    public void createGoal(String name, Duration targetDuration) {
        gateway.createGoal(name, targetDuration).thenAccept(result -> {
            _actionResult.postValue(result);
            // 목표 생성 성공 시 목록 새로고침
            if (result instanceof ActionResult.Success) {
                loadGoals();
            }
        });
    }
}

File: src/main/java/com/timeback/ui/feature/goals/GoalsViewModel.kt
package com.timeback.ui.feature.goals

import androidx.lifecycle.ViewModel
import androidx.lifecycle.viewModelScope
import com.timeback.ui.domain.gateway.FeatureGateway
import com.timeback.ui.domain.model.*
import dagger.hilt.android.lifecycle.HiltViewModel
import kotlinx.coroutines.flow.MutableStateFlow
import kotlinx.coroutines.flow.StateFlow
import kotlinx.coroutines.flow.asStateFlow
import kotlinx.coroutines.launch
import javax.inject.Inject

/**
 * UI-06 목표 ViewModel
 * CT-04 §9: ReadGoals, CreateGoal
 */
@HiltViewModel
class GoalsViewModel @Inject constructor(
    private val gateway: FeatureGateway
) : ViewModel() {

    private val _state = MutableStateFlow<ScreenState>(ScreenState.Initial)
    val state: StateFlow<ScreenState> = _state.asStateFlow()

    private val _actionResult = MutableStateFlow<ActionResult?>(null)
    val actionResult: StateFlow<ActionResult?> = _actionResult.asStateFlow()

    init {
        loadGoals()
    }

    fun loadGoals() {
        viewModelScope.launch {
            _state.value = ScreenState.Loading
            _state.value = gateway.readGoals()
        }
    }

    fun createGoal(name: String, targetMinutes: Long) {
        viewModelScope.launch {
            _state.value = ScreenState.Loading
            val result = gateway.createGoal(name, Duration(targetMinutes * 60_000))
            _actionResult.value = result
            if (result is ActionResult.Success) {
                _state.value = gateway.readGoals()
            }
        }
    }

    fun clearActionResult() {
        _actionResult.value = null
    }
}
